crates/ironrdp-pdu/src/rdp/capability_sets/general.rs

```diff
@@ -111,7 +111,7 @@ impl PduParsing for General {
 
         let protocol_ver = buffer.read_u16::<LittleEndian>()?;
         if protocol_ver != PROTOCOL_VER {
-            return Err(CapabilitySetsError::InvalidProtocolVersion);
+            warn!("Invalid protocol version: {protocol_ver:?}");
         }
```

Comment:
You’re right that this should not be a fatal error, but I feel this check should not exist in `ironrdp-pdu` at all in the first place. I would see it somewhere in `ironrdp-connector` / `ironrdp-acceptor`. As much as possible, I want the `ironrdp-pdu` crate to focus more on decoding / encoding PDUs in a lenient way and hold less such logic, leaving it to high layers.

Reply:
ok, I added a protocol_version field to General, and an "impl Default", should be enough for now.